src/app/components/tables/data-table/terra-data-table.component.spec.ts

```diff
@@ -0,0 +1,122 @@
+
+describe('Component: TerraDataTableComponent', () =>
+{
+    let component:TerraDataTableComponent<any, any>; // TODO T and P should be defined
+    let fixture:ComponentFixture<TerraDataTableComponent<any, any>>;
+
+    beforeEach(async(() =>
+    {
+        TestBed.configureTestingModule({
+            declarations: [
+                TerraDataTableComponent,
+                TerraButtonComponent,
+                TerraPagerComponent,
+                TerraCheckboxComponent,
+                TerraDataTableContextMenuComponent,
+                TerraDataTableContextMenuDirective,
+                TerraNoResultNoticeComponent,
+                TerraTaglistComponent,
+                TerraTagComponent,
+                TerraBaseToolbarComponent,
+                TerraNumberInputComponent,
+                TerraSelectBoxComponent
+            ],
+            imports:      [
+                TooltipModule.forRoot(),
+                CommonModule,
+                FormsModule,
+                HttpModule,
+                HttpClientModule,
+                NoopAnimationsModule,
+                LocalizationModule.forRoot(l10nConfig)
+            ],
+            providers:    [
+                TerraDataTableServiceExample,
+                TerraLoadingSpinnerService
+            ]
+        }).compileComponents();
+    }));
+
+    beforeEach(() =>
+    {
+        fixture = TestBed.createComponent(TerraDataTableComponent);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
+    });
+
+    it('should create', () =>
+    {
+        expect(component).toBeTruthy();
+    });
+
+    describe('#inputHasPager', () =>
+    {
+        it('should initialise `inputHasPager` with true', () =>
+        {
+            expect(component.inputHasPager).toBe(true);
+        });
+
+        it('should hide the pager if `inputHasPager` is set but no data is available', () =>
+        {
+            let service:TerraDataTableServiceExample = TestBed.get(TerraDataTableServiceExample);
+            let pagerDE:DebugElement = fixture.nativeElement.querySelector('terra-pager');
+            expect(component.inputHasPager).toBe(true);
+            expect(service.rowList).toBeDefined();
+            expect(service.rowList.length).toEqual(0);
+            expect(pagerDE).toBeNull();
+        });
+
+        // TODO: Make it run!
+        xit('should have a pager if `inputHasPager` is set and data is available', async(() =>
+        {
+            let service:TerraDataTableServiceExample = TestBed.get(TerraDataTableServiceExample);
+            let pagerDE:DebugElement = fixture.nativeElement.querySelector('terra-pager');
+            service.getResults();
+            fixture.detectChanges();
+            expect(service.rowList).toBeDefined();
+            expect(service.rowList.length).toBe(3);
+            expect(pagerDE).toBeTruthy();
+        }));
+    });
+
+    describe('With an inputService', () =>
```

Comment:
kannste das nicht mit ersten beforeEach machen?

Reply:
Kann ich. Aber so gilt das nur für die Tests in dem describe block